docker/README.md

```diff
@@ -0,0 +1,47 @@
+This document is used for generating OWT build and all-in-one OWT running environment images.
+
+## Prequisites
+
+  * Docker is setup in your local environment and related proxy is set so that Docker can access to external network (including Google because OWT will download WebRTC related code and patch from Google)
+
+## Build OWT images
+
+1. Run following command to generate Docker image containing OWT source code and dependencies. A dist folder will be generated with all OWT running binaries. You can run OWT service using this image directly in case you want to modify OWT source code and debug.
+
+    ```shell
+    docker build --target owt-build -t owt:build \
+            --build-arg http_proxy=${HTTP_PROXY} \
+            --build-arg https_proxy=${HTTPS_PROXY} \
+            .
+    ```
+
+2. Run following command to generate OWT running environment Docker image with OWT binaries generated in step 1, not including source code.
+    ```shell
+    docker build --target owt-run -t owt:run \
+    --build-arg http_proxy=${HTTP_PROXY} \
+    --build-arg https_proxy=${HTTPS_PROXY} \
+    .
+    ```
+Or you can simply run `build_docker_image.sh` to generate both 2 images.
+We provide a `startowt.sh` to launch OWT and expose following parameters to configure:
+
+```
+    --rabbit rabbitmq server address if rabbitmq service is deployed on a different device
+    --mongo mongodb server address if mongo service is deployed on a different device, like --mongo=xx.xx.xx.xx/owtdb
+    --externalip external ip of host device if there are internal and external ip for host device
+    --network_interface network interface for external ip of host device if there are external and internal ip for host device
+```
+Note that `externalip` and `network_interface` should be both set if there are external and internal ip on host device. `startowt.sh` is just a reference, you can customize Dockerfile and script to generate the Docker images for cluster deployment. For example, separate each OWT modules into different Docker images.
+
+2. Launch OWT service
+
+    a) Launch a Docker container with following command to simply try OWT:
+
+    ```shell
+    docker run -itd --name=owt --net=host owt:run /home/startowt.sh
```

Comment:
mongodb and rabbitmq won't be started though, right?

Reply:
we install mongod and rabbitmq inside the Docker image so mongo and rabbitmq will be started automatically inside the container, user can choose not start mongod and rabbitmq service and use their own service

Reply:
I don't think they are started automatically; when I try the command above it doesn't work. If I start the container manually it doesn't work. Because init-all hasn't been run I think. Sorry, I still haven't really given this much time.

Reply:
The mongod and rabbitmq service launch action is missing in the script, will update the script, thanks for pointing it out

Reply:
I see mongo/rabbitmq are started now, however I get a ECONNREFUSED when cluster-manager starts up, I think because RabbitMQ isn't really ready yet. So things don't really all come up.

I remember you had some script to wait for MongoDB / RabbitMQ to really be running successfully, maybe that should go here?

Reply:
ECONNREFUSED is caused that rabbitmq service isn't ready, and OWT modules will try to connect to rabbitmq. I will add mongodb service launch status check in the script for OWT will only try to connect to mongod once. Please review.

Reply:
I still see that ECONNREFUSED most of the time. I submitted a PR to your branch that fixes it, if you like.

Reply:
Thanks for the comment, i just updated it to this PR, please review.